mathesar/api/serializers/queries.py

```diff
@@ -29,8 +30,29 @@ def validate(self, attrs):
         unexpected_fields = set(self.initial_data) - set(self.fields)
         if unexpected_fields:
             raise ValidationError(f"Unexpected field(s): {unexpected_fields}")
+        self._validate_uniqueness(attrs)
         return attrs
 
+    def _validate_uniqueness(self, attrs):
+        """
+        Uniqueness is only defined when both name and base_table are defined.
+
+        Would be nice to define this in terms of Django's UniqueConstraint, but that doesn't seem
+        possible, due to schema being a child property of base_table.
```

Comment:
Argh, that sucks. I think you're right, though. Maybe that's why we went with unique constraining names in the first place.